Add dast documentation [TAROT-3247] (#2371)
codacy/docs: docs/codacy-api/examples/running-dast-scans.md

---
description: Instructions on how to run DAST/App using the API.
---

# Run DAST/App scans

Thanks to the new app scanning capabilities available on the Security and Risk Management dashboard, it's now possible to automate application scanning via the API. This means that, with little effort, you'll be able to run ZAP on demand every time you deploy a new version of your app or your APIs.

!!! important
    App scanning is a business feature. If you are a Codacy Pro customer, contact our customer success team to access a short trial.

>     Check your permissions. Only git provider admins and organization managers will be able to Create new targets and run scans (in app and via the API).

Reviewer: We should add these permissions here: https://docs.codacy.com/organizations/roles-and-permissions-for-organizations/

And link that page here.

Replace the placeholders with your own values:

-   **API_KEY:** [Account API token](../api-tokens.md#account-api-tokens) used to authenticate on the Codacy API.
-   **GIT_PROVIDER:** Git provider hosting of the organization, using one of the values in the table below. For example, `gh` for GitHub Cloud.

    | Value | Git provider      |
    |-------|-------------------|
    | `gh`  | GitHub Cloud      |
    | `ghe` | GitHub Enterprise |
    | `gl`  | GitLab Cloud      |
    | `gle` | GitLab Enterprise |
    | `bb`  | Bitbucket Cloud   |
    | `bbe` | Bitbucket Server  |

-   **ORGANIZATION:** Name of the organization on the Git provider. For example, `codacy`. You must have admin permissions over the organization on the Git provider.
-   **DAST_TARGET_ID:** Identifier of a DAST target to analyze (obtained from the POST endpoint used to create a target or GET endpoint used to get a list of targets). For example, `457`. You must have admin permissions over the organization on the Git provider.

Scans will be asynchronous to your deployment pipeline and the results will be posted within the SRM dashboard within the findings tab.

## Examples to Run scans on Deployment

!!! note

>     The examples below trigger scans on deployment which may analyze, not the latest, but the previous version of the application. To run DAST post deployment you may explore options such as:

Reviewer 2: I was trying to give options to users here, but honestly I don't really know what to suggest... 

Ideally, the examples we had below would work post deploy (the one from circleci seems to work for example), but that doesn't seem the case. @lventura-codacy @afsmeira @lorandszakacs could any of your gentle souls take a look at these examples and review / improve them? 😭 thank u guys

Reviewer: I would honestly not provide these examples and I wouldn't have this section. This is the type of docs that get outdated quickly and are high maintenance.

I would just mention that the natural way to use DAST scanning is to trigger a scan after the application you want to scan is deployed to it's production environment.

To people that use CI/CD pipelines to do these deployments, it should be trivial to add an HTTP call afterwards and check the status code to see if it suceeded or not.

People that don't use it need to call it manually or use the UI.

Author: are we ok deleting this? @claudiacodacy

Reviewer 2: yesssssss <3

---
description: Instructions on how to run DAST/App using the API.
---

# Run Dynamic Application Security Testing (DAST) scans

Thanks to the new app scanning capabilities available on the Security and risk management dashboard, it's now possible to automate application scanning via Codacy's API. This means that, with little effort, you'll be able to run app scanning on demand every time you deploy a new version of your app.

!!! important
    App scanning is a business feature. If you are a Codacy Pro customer, contact our customer success team to access a short trial.

>     Check your permissions. Only git provider admins and organization managers will be able to create new targets and run scans (in app and via the API).

Reviewer: Link to permissions page.

To create a target:

```bash
curl -X POST https://app.codacy.com/api/v3/organizations/{GIT_PROVIDER}/{ORGANIZATION}/dast/targets \
  -H "Content-Type: application/json" \
  -H "Accept: application/json" \
  -H "api-token: {API_KEY}" \
  -d '{"url": "https://api.domain.com/v1"}'
```

Replace the placeholders with your own values:

-   **API_KEY:** [Account API token](../api-tokens.md#account-api-tokens) used to authenticate on the Codacy API.
-   **GIT_PROVIDER:** Git provider hosting of the organization, using one of the values in the table below. For example, `gh` for GitHub Cloud.

    | Value | Git provider      |
    |-------|-------------------|
    | `gh`  | GitHub Cloud      |
    | `ghe` | GitHub Enterprise |
    | `gl`  | GitLab Cloud      |
    | `gle` | GitLab Enterprise |
    | `bb`  | Bitbucket Cloud   |
    | `bbe` | Bitbucket Server  |

-   **ORGANIZATION:** Name of the organization on the Git provider. For example, `codacy`. You must have admin permissions over the organization on the Git provider.

Once you create the target you'll get the 'Target Id' as a response. That id will be used to trigger DAST scans in the next section.

> ## Running DAST analysis scans

Reviewer: It's funny that we mention 'trigger DAST scans in the next section' and then use the word running. I think we're using a bit of both and should stick to one nomenclature.

Once your targets are created you can trigger an analysis by calling the '[Analyze DAST target](https://api.codacy.com/api/api-docs#analyzedasttarget)' endpoint.

```bash
curl -X POST https://app.codacy.com/api/v3/organizations/{GIT_PROVIDER}/{ORGANIZATION}/dast/targets/{DAST_TARGET_ID}/analyze \
  -H "Accept: application/json" \
  -H "api-token: {API_KEY}"
```

Replace the placeholders with your own values:

-   **API_KEY:** [Account API token](../api-tokens.md#account-api-tokens) used to authenticate on the Codacy API.
-   **GIT_PROVIDER:** Git provider hosting of the organization, using one of the values in the table below. For example, `gh` for GitHub Cloud.

    | Value | Git provider      |
    |-------|-------------------|
    | `gh`  | GitHub Cloud      |
    | `ghe` | GitHub Enterprise |
    | `gl`  | GitLab Cloud      |
    | `gle` | GitLab Enterprise |
    | `bb`  | Bitbucket Cloud   |
    | `bbe` | Bitbucket Server  |

-   **ORGANIZATION:** Name of the organization on the Git provider. For example, `codacy`. You must have admin permissions over the organization on the Git provider.

> -   **DAST_TARGET_ID:** Identifier of a DAST target to analyze (obtained from the POST endpoint used to create a target or GET endpoint used to get a list of targets). For example, `457`. You must have admin permissions over the organization on the Git provider.

Reviewer: It's the first time we talk about the list targets endpoint... I would avoid it.

```suggestion
-   **DAST_TARGET_ID:** Identifier of a DAST target to analyze (obtained in the [previous section](LINK TO CORRECT SECTION)). For example, `457`. You must have admin permissions over the organization on the Git provider.
```

> -   **DAST_TARGET_ID:** Identifier of a DAST target to analyze (obtained from the POST endpoint used to create a target or GET endpoint used to get a list of targets). For example, `457`. You must have admin permissions over the organization on the Git provider.

Reviewer 2: ```suggestion
-   **DAST_TARGET_ID:** Identifier of a DAST target to analyze (obtained from the POST endpoint used to create a target or [GET endpoint](https://api.codacy.com/api/api-docs#getdasttargets) used to get a list of targets). For example, `457`. You must have admin permissions over the organization on the Git provider.
```
